### PR TITLE
fix(rcm): fix release note and validation logic from #4970

### PR DESCRIPTION
**Pull request by avara1986**
## Description

This PR contains two changes to the #4970 PR:

- Update the release note to clarify the scope + bug being fixed
- Rollback incorrect validation based on feedback from https://github.com/DataDog/dd-trace-py/pull/4974#discussion_r1089321493

### CODE REVIEW

**Comment by emmettbutler** [comment]
Can you provide any additional context for this change?

**Review by gnufede — DISMISSED** [review]

**Comment by gnufede** [comment]
> Can you provide any additional context for this change?
https://github.com/DataDog/dd-trace-py/pull/4974#discussion_r1089321493

**Review by emmettbutler — DISMISSED** [review]

File: ddtrace/internal/remoteconfig/client.py [hunk 1]
```diff
@@ -348,7 +348,8 @@ def _process_response(self, data):
         paths = {_.path for _ in payload.target_files}
         paths = paths.union({_["path"] for _ in self.cached_target_files})
 
-        if set(payload.client_configs) > paths:
+        # !(payload.client_configs is a subset of paths or payload.client_configs is equal to paths)
+        if not set(payload.client_configs) <= paths:
```
Comment (brettlangdon) [inline ddtrace/internal/remoteconfig/client.py:352]: I'll update the release note as well.

**Review by brettlangdon — APPROVED** [review]

File: releasenotes/notes/rcm-validations-162716c50ea90f1b.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 ---
 fixes:
   - |
-    Add missing validation to Remote Config response from agent.
\ No newline at end of file
+    Fix incorrect target file validation of Remote Config responses from the agent.
```
Comment (brettlangdon) [inline releasenotes/notes/rcm-validations-162716c50ea90f1b.yaml:4]:
```suggestion
    remote config: ensure proper validation of responses from the agent.
```

**Review by brettlangdon — APPROVED** [review]

**Review by gnufede — APPROVED** [review]

**Pull request merged**